posthog/temporal/proxy_service/delete.py

```diff
@@ -0,0 +1,133 @@
+from asgiref.sync import sync_to_async
+from dataclasses import dataclass
+import datetime as dt
+import grpc.aio
+import json
+import uuid
+
+from temporalio import activity, workflow
+import temporalio.common
+
+from posthog.models import ProxyRecord
+from posthog.temporal.batch_exports.base import PostHogWorkflow
+from posthog.temporal.common.logger import bind_temporal_org_worker_logger
+
+from posthog.temporal.proxy_service.common import (
+    get_grpc_client,
+    NonRetriableException,
+    update_proxy_record,
+    UpdateProxyRecordInputs,
+)
+from posthog.temporal.proxy_service.proto import DeleteRequest
+
+
+@dataclass
+class DeleteProxyRecordInputs:
+    organization_id: uuid.UUID
+    proxy_record_id: uuid.UUID
+
+
+@dataclass
+class DeleteHostedProxyInputs:
+    """Inputs for the DeleteHostedProxy Workflow and Activity."""
+
+    organization_id: uuid.UUID
+    proxy_record_id: uuid.UUID
+    domain: str
+
+
+@activity.defn
+async def delete_proxy_record(inputs: DeleteProxyRecordInputs):
+    """Activity that does a DNS lookup for the target subdomain and checks it has a CNAME
+    record matching the expected value.
+    """
+    logger = await bind_temporal_org_worker_logger(organization_id=inputs.organization_id)
+    logger.info(
+        "Deleting proxy record %s",
+        inputs.proxy_record_id,
+    )
+
+    pr = await sync_to_async(ProxyRecord.objects.get)(id=inputs.proxy_record_id)
+    await sync_to_async(pr.delete())
```

Comment:
```suggestion
    await sync_to_async(pr.delete)()
```

Reply:
ooooh that's what the time error was from 🤦

Reply:
well, my change fixed it anyway and i think it's more readable